lib/opte/src/ddi/sync.rs

```diff
@@ -118,7 +118,14 @@ impl<T> KMutex<T> {
         //
         // Safety: ???.
         unsafe {
-            mutex_init(&mut kmutex, ptr::null(), mtype.into(), ptr::null());
+            // MUTEX_DRIVER is the only type currently sanctioned by the DDI
+            // for use here. Other types require the priority argument to be set.
```

Comment:
`MUTEX_DRIVER` is the only sanctioned type, but you can use `MUTEX_DRIVER ` along with a priority argument when creating a mutex for use inside an interrupt handler.  In that case you may well end up with a spin lock depending on the priority.

This is what the comment at 114:115 is alluding to. This'll need to grow support for the priority argument if you ever want interrupt context locks, but you'd still be using `MUTEX_DRIVER` for the type.